### PR TITLE
ISSUE #2640: BP-43 Migrate bookkeeper client tests to gradle

### PR DESCRIPTION
**Pull request by pkumar-singh**
### Motivation
Sometimes ago gradle was introduced to Bookkeeper, 
and with that build and release capabilities were built in Bookkeeper with gradle. But we never migrated bookkeeper to gradle.

Taking a first step towards gradle migration; This PR migrate all client unit tests of bookkeepr-server module to gradle 
### Changes
Replace maven test target with gradle

Master Issue: #2640

### CODE REVIEW

File: bookkeeper-server/build.gradle [hunk 1]
```diff
@@ -18,6 +18,8 @@
  */
 plugins {
     id 'java'
+    id 'com.adarshr.test-logger'
+    //id 'org.gradle.test-retry'
```
Comment (merlimat) [inline bookkeeper-server/build.gradle:22]: we can remove if not used for now

**Review by merlimat — APPROVED** [review]

File: bookkeeper-server/build.gradle (hunk 1 above)
Comment (pkumar-singh) [reply bookkeeper-server/build.gradle:22]: yeah

**Review by hsaputra — APPROVED** [review]
+1

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
It's better to remove 'test-retry' unused stuff

**Comment by hsaputra** [comment]
Will merge this by end of day if no more comment or suggestion. Thanks

**Comment by pkumar-singh** [comment]
@nicoloboschi I will uncomment that in next commit, If you do not mind. It will follow soon.

File: bookkeeper-server/build.gradle (hunk 1 above)
Comment (hsaputra) [reply bookkeeper-server/build.gradle:22]: @pkumar-singh - could you update the PR to remove at least this line?

Comment (pkumar-singh) [reply bookkeeper-server/build.gradle]: Sure done.

**Comment by pkumar-singh** [comment]
> @nicoloboschi I will uncomment that in next commit, If you do not mind. It will follow soon.

Done.

**Comment by nicoloboschi** [comment]
@pkumar-singh thanks, i mean to remove it also from properties and settings, but we can keep them if they will be used in the next PR

**Review by nicoloboschi — APPROVED** [review]
+1

**Comment by hsaputra** [comment]
rerun failure checks

**Comment by hsaputra** [comment]
@merlimat - could you help merge this one? I get this error:

```
Unable to fetch URL, exiting: https://api.github.com/repos/apache/bookkeeper/pulls/2800/merge - HTTP Error 404: Not Found
HTTP Error 404: Not Found
```
again =(

Thanks

**Comment by hsaputra** [comment]
Duplicated by https://github.com/apache/bookkeeper/pull/2807 to help with the merging

**Pull request closed without merging**